internal/services/bot/bot_resource.go

```diff
@@ -0,0 +1,313 @@
+package bot
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/botservice/mgmt/2021-03-01/botservice"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/location"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/bot/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceBot() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceBotCreate,
+		Read:   resourceBotRead,
+		Update: resourceBotUpdate,
+		Delete: resourceBotDelete,
+		Importer: pluginsdk.ImporterValidatingResourceIdThen(func(id string) error {
```

Comment:
In this particular case I think the regular Importer function `pluginsdk.ImporterValidatingResourceId` would suffice here. Typically the custom importers are used to check the type (in this case the kind) of bot that's being imported. But since this resource supports all the kinds this check doesn't need to be performed.

Reply:
Thanks! I updated the code to reflect the suggestion here.